services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/protocol/serializer/SchemaDiffTraverser.java

```diff
@@ -0,0 +1,300 @@
+package com.linkedin.venice.controller.kafka.protocol.serializer;
+
+import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper;
+import com.linkedin.venice.utils.AvroSchemaUtils;
+import com.linkedin.venice.utils.Pair;
```

Comment:
Note that `Pair` is deprecated and we prefer using dedicated classes instead. There is a similar class which already exists called `SchemaPairAndClassContainer`, though it contains one extra field that you likely don't need (though it supports being null, so technically it would still work...). You can also create a new `SchemaReaderWriterPair` class, or similar.